conorfirebasewk2/app/src/main/java/com/example/conor_firebase_wk2/NewPost.java

```diff
@@ -1,27 +1,45 @@
 package com.example.conor_firebase_wk2;
 
+import androidx.annotation.NonNull;
 import androidx.appcompat.app.AppCompatActivity;
 
+import android.app.ProgressDialog;
 import android.os.Bundle;
 import android.view.View;
 import android.widget.EditText;
+import android.widget.Toast;
 
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.Task;
 import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.firestore.FirebaseFirestore;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
 
 public class NewPost extends AppCompatActivity implements View.OnClickListener{
 
     private EditText userPost;
     private FirebaseAuth mAuth;
 
-    @Override
+    ProgressDialog pd;
+
+    FirebaseFirestore db;
+
+
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_new_post);
 
 
-        userPost = findViewById(R.id.postContent);
+        pd = new ProgressDialog(this);
 
-        mAuth = FirebaseAuth.getInstance();
+        db = FirebaseFirestore.getInstance();
```

Comment:
this can be all initialised at the top where you have created the name of the object..if you get me